broadinstitute/scp-ingest-pipeline
# Finish integrating author DE into Ingest Pipeline

Extend implementation for author DE into the Ingest Pipeline codebase, by adding it to ingest_pipeline.py and adding a new parser to cli_parser.py. The structure of doing so is very similar to de.py and how automated DE files are generated and pushed to google bucket from the ingest pipeline. The end result is individual comparison files being written, for pairwise and rest DE, and uploaded to google bucket. See an example bucket [here](https://console.cloud.google.com/storage/browser/fc-3dab559a-a5ce-42a6-96e7-1e04228c10b8/_scp_internal/differential_expression;tab=objects?pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22,%22s%22:%5B(%22i%22:%22objectListDisplayFields%2FtimeLastModified%22,%22s%22:%221%22),(%22i%22:%22displayName%22,%22s%22:%220%22)%5D))&prefix=&forceOnObjectsSortingFiltering=true). 

### Test 

1.  Go to your local SCP
2. Find "Human milk - differential expression" study
3. Go to its Google bucket
4. Upload the example author DE file from `scp-ingest-pipeline/tests/data/author_de/author_de_test_data_human_milk_All_Cells_UMAP_General_celltype.csv` to the top-level GCS bucket directory
5. Run the following command, replacing `fc-febd4c65-881d-497f-b101-01a7ec427e6a` with your bucket ID.

```
python ingest_pipeline.py --study-id addedfeed000000000000000 --study-file-id dec0dedfeed1111111111111 ingest_differential_expression --annotation-name General_Celltype --annotation-type group --annotation-scope study --cluster-name cluster_umap_txt --study-accession SCPdev --ingest-differential-expression --differential-expression-file gs://fc-febd4c65-881d-497f-b101-01a7ec427e6a/author_de_test_data_human_milk_All_Cells_UMAP_General_celltype.csv --method wilcoxon
```

6.  Back in your GCS bucket, confirm you have new files in `_scp_internal/differential_expression/`.
